File: NeetCode/BinarySearch/Easy/977.SquaresSortedArray.py
# Given an integer array nums sorted in non-decreasing order, 
# return an array of the squares of each number sorted in non-decreasing order.

def sortedSquares(nums):
    left, right = 0, len(nums) - 1
    result = []

    while left <= right:
        if nums[left]**2 >= nums[right]**2:
            result.append(nums[left]**2)
            left += 1
        else:
            result.append(nums[right]**2)
            right -= 1

    return result[::-1]

print(sortedSquares([-4,-1,0,3,10]))
print(sortedSquares([-7,-3,2,3,11]))

File: swift/Neetcode/BinarySearch/Easy/977.SquaresSortedArray.swift
// Given an integer array nums sorted in non-decreasing order, 
// return an array of the squares of each number sorted in non-decreasing order.

func sortedSquares(_ nums: [Int]) -> [Int] {
    var left = 0, right = nums.count - 1
    var result = [Int]()

    while left <= right {
        if abs(nums[left]) >= abs(nums[right]) {
            result.append(nums[left] * nums[left])
            left += 1
        } else {
            result.append(nums[right] * nums[right])
            right -= 1
        }
    }

    return result.reversed()
}

print(sortedSquares([-4,-1,0,3,10]))
print(sortedSquares([-7,-3,2,3,11]))